app/models/user.py

```diff
@@ -185,6 +205,40 @@ def is_admin(self):
         return False
 
 
+class Vendor(User):
+    # __tablename__ = 'vendors'
+    __mapper_args__ = {'polymorphic_identity': 'vendor'}
+    id = db.Column(db.Integer, db.ForeignKey('users.id'), primary_key=True)
+
+    # are the vendor's prices visible to other vendors?
+    visible = db.Column(db.Boolean, default=False)
+
+    # TODO: one-to-many relationships to LISTINGs
+
+    def __init__(self, **kwargs):
+        super(Vendor, self).__init__(**kwargs)
+        self.visible = kwargs.get('visible', False)
+        self.role = Role.query.filter_by(index='vendor').first()
+
+    def __repr__(self):
+        return '<Vendor %s>' % self.full_name()
+
+
+class Merchant(User):
+    # __tablename__ = 'merchants'
```

Comment:
When I had separate table names, I kept getting this issue of creating a user in one table without it populating it's entry in the inherited table.